File: src/cn/edustar/jitar/service/UserSubjectGradeQuery.java
package cn.edustar.jitar.service;

import cn.edustar.jitar.data.BaseQuery;
import cn.edustar.jitar.data.QueryContext;

public class UserSubjectGradeQuery extends BaseQuery {

	public Integer subjectId = null;
	public Integer gradeId = null;
	public Integer userId = null;
	// 是否模糊匹配
	public boolean FuzzyMatch = false;
	public Integer orderType = 0;

	public UserSubjectGradeQuery(String selectFields) {
		super(selectFields);
	}

	
	@Override
	public void initFromEntities(QueryContext qctx) {
		qctx.addEntity("UserSubjectGrade", "usg", "");
	}

	@Override
	public void applyWhereCondition(QueryContext qctx) {
		if (subjectId != null) {
			qctx.addAndWhere("usg.subjectId = :subjectId");
			qctx.setInteger("subjectId", subjectId);
		}

		if (gradeId != null) {
			if (!FuzzyMatch) {
				qctx.addAndWhere("usg.gradeId = :gradeId");
				qctx.setInteger("gradeId", gradeId);
			} else {
				qctx.addAndWhere("usg.gradeId >= :gradeStartId And usg.gradeId < :gradeEndId");
				qctx.setInteger("gradeStartId", convertRoundMinNumber(gradeId));
				qctx.setInteger("gradeEndId", convertRoundMaxNumber(gradeId));
			}

		}

		if (userId != null) {
			qctx.addAndWhere("usg.userId = :userId");
			qctx.setInteger("userId", userId);
		}

	}

	@Override
	public void applyOrderCondition(QueryContext qctx) {
		 if (orderType == 0){
			 qctx.addOrder("usg.userSubjectGradeId DESC");
		 }
	}

	@Override
	public void resolveEntity(QueryContext qctx, String entity) {
		super.resolveEntity(qctx, entity);
	}

	
	public void setSubjectId(Integer subjectId) {
		this.subjectId = subjectId;
	}

	public void setGradeId(Integer gradeId) {
		this.gradeId = gradeId;
	}

	public void setUserId(Integer userId) {
		this.userId = userId;
	}

	public void setFuzzyMatch(boolean fuzzyMatch) {
		FuzzyMatch = fuzzyMatch;
	}

	public void setOrderType(Integer orderType) {
		this.orderType = orderType;
	}

}

File: WebContent/WEB-INF/jython/user_subject_grade_query.py
# -*- coding: utf-8 -*-
from cn.edustar.jitar.data import BaseQuery
from base_action import SubjectMixiner

class UserSubjectGradeQuery(BaseQuery, SubjectMixiner):
    def __init__(self, selectFields):        
        BaseQuery.__init__(self, selectFields)
        self.subjectId = None
        self.gradeId = None
        self.userId = None
        self.FuzzyMatch = None
        self.orderType = 0
        
    def initFromEntities(self, qctx):
        qctx.addEntity("UserSubjectGrade" , "usg", "")
    
    def resovleEntity(self, qctx, entity):
        BaseQuery.resolveEntity(self, qctx, entity)
    
    def applyWhereCondition(self, qctx):
        if self.subjectId != None:
            qctx.addAndWhere("usg.subjectId = :subjectId")
            qctx.setInteger("subjectId", self.subjectId)
         
        if self.gradeId != None:
            if self.FuzzyMatch == None or self.FuzzyMatch == False:           
                qctx.addAndWhere("usg.gradeId = :gradeId")
                qctx.setInteger("gradeId", self.gradeId)
            else:
                qctx.addAndWhere("usg.gradeId >= :gradeStartId And usg.gradeId < :gradeEndId")
                qctx.setInteger("gradeStartId", self.calcGradeStartId(self.gradeId))
                qctx.setInteger("gradeEndId", self.calcGradeEndId(self.gradeId))
                
        if self.userId != None:
            qctx.addAndWhere("usg.userId = :userId")
            qctx.setInteger("userId", self.userId)
            
    def applyOrderCondition(self, qctx):
        # 排序方式, 0 - placardId DESC
        if self.orderType == 0:
            qctx.addOrder("usg.userSubjectGradeId DESC")